### ISSUE TITLE
Issue with "Array.prototype.reduce()": (conflicting info on usage)

### ISSUE THREAD

**Issue by ddavies-integral** [mdn/content#6156]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

#### What information was incorrect, unhelpful, or incomplete?

In the Syntax section it describes the callbackFn as '_A function to execute on each element in the array (except for the first, if no initialValue is supplied)_.'. It's saying that if you don't provide an initialValue param and your array has items 0, 1, 2, and 3 then it will execute only three times - on indexes 1, 2, and 3. It will skip 0. It explicitly states this when describing the index:
_'The index of the current element being processed in the array. Starts from index 0 if an initialValue is provided. Otherwise, it starts from index 1'_.

#### Specific section or headline?

Syntax

#### What did you expect to see?

I was perfectly happy to accept this behaviour as described, but (see test section):

#### Did you test this? If so, how?

The page has it's own executor playground area and has code that contradicts the expected behaviour. I simply ran that code and it ran (WITHOUT initialValue and therefore should skip array element zero) contradictarily (new word, (c) me). It ran through the whole array, including element zero despite no initial value and accumulated array elements 0-3 giving a total value of 10. It would be 9 if the first element was skipped. Here's the code:

```
const array1 = [1, 2, 3, 4];
const reducer = (accumulator, currentValue) => accumulator + currentValue;

// 1 + 2 + 3 + 4
console.log(array1.reduce(reducer));     <<< no initial value supplied
// expected output: 10                           <<< but the expected output is somehow ten, why not 9?
```

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/array/reduce`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/array/reduce/index.html
* Last commit: https://github.com/mdn/content/commit/761ba0efe8230cec103437c215a79a2e6d2bb5f2
* Document last modified: 2021-06-15T08:30:35.000Z

</details>

**Comment by sideshowbarker** [comment]
For the *initialValue*, the Syntax section says this:

> If no `initialValue` is supplied, the first element in the array will be used as the initial accumulator value and skipped as `currentValue`.

So per that, and given this:

```js
const array1 = [1, 2, 3, 4];
const reducer = (accumulator, currentValue) => accumulator + currentValue;

// 1 + 2 + 3 + 4
console.log(array1.reduce(reducer));
// expected output: 10
```
…then what happens is:

1. `accumulator` is initialized to `1` (the first element in the array).
2. `reducer` runs with `currentValue` set to `2`, so the results is `accumulator = 1 + 2`, so `3`.
3. `reducer` runs with `currentValue` set to `3`, so the results is `accumulator = 3 + 3`, so `6`.
4. `reducer` runs with `currentValue` set to `4`, so the results is `accumulator = 6 + 4`, so `10`.

So, to be clear, the first element if the array isn’t just skipped or ignored — instead it’s *always* used — it’s just that if no other `initialValue` is supplied, then the first element of the array is  used as the initial value.

In other words, the accumulator is *never* initialized to `0` unless either: `0` is the first element in the array you provide, or otherwise if you explicitly supply an `initialValue` argument with the value `0`.

So can you suggest any change to the current wording of the article which would make the above more clear?

**Comment by ddavies-integral** [comment]
Yeah, if you read the entire thing from start to finish, it does become apparent that this is what's happening. But if you isolate your reading to just one or two fields, then it's less obvious. My only suggestion would be that in each case where it says that it's "starting at 1", it might also mention "since the zero index value is the default initializer of the accumulator".

**Comment by sideshowbarker** [comment]
I’ve opened #7484 with a fix for this.
